tests/integration/states/test_ssh_auth.py

```diff
@@ -102,3 +102,42 @@ def test_issue_10198_keyfile_from_another_env(self, username=None):
             self.assertEqual(
                 fhr.read(), "ssh-rsa AAAAB3NzaC1kcQ9J5bYTEyZ== {}\n".format(username)
             )
+
+    @pytest.mark.destructive_test
+    @with_system_user("issue_60769", on_existing="delete", delete=True)
+    @pytest.mark.slow_test
+    @pytest.mark.skip_if_not_root
+    def test_issue_60769_behavior_using_source_and_options(self, username=None):
+        userdetails = self.run_function("user.info", [username])
+        user_ssh_dir = os.path.join(userdetails["home"], ".ssh")
+        authorized_keys_file = os.path.join(user_ssh_dir, "authorized_keys")
+        pub_key_file = "issue_60769.id_rsa.pub"
+
+        # create a prepared authorized_keys file with option
+        try:
+            os.mkdir(user_ssh_dir)
+        except FileExistsError:
+            print("folder {} already exists".format(user_ssh_dir))
```

Comment:
Was the print for your own testing? Probably could use a pass here instead, or log to debug using log.debug("msg") instead. regular print won't show up well in our log

Reply:
Alternatively a check to see if the directory exists and if not, then create it.

Reply:
Ok indeed, print was for debugging, I've updated the branch with `log.debug`.